File: examples/java/Search2.java
public class Search2 {
    static int binarySearch(int[] arr, int key) {
        int lo = 0, hi = arr.length - 1;
        while (lo <= hi) {
            int mid = lo + (hi - lo) / 2;
            if (arr[mid] == key) return mid;
            if (arr[mid] < key) lo = mid + 1;
            else hi = mid - 1;
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] arr = {2, 5, 8, 12, 16, 23, 38};
        int key = 12;
        int idx = binarySearch(arr, key);
        System.out.println(idx >= 0 ? "Found at " + idx : "Not found");
    }
}

File: examples/csharp/Search2.cs
using System;

class Search2
{
    static void Main()
    {
        int[] arr = { 1, 3, 5, 7, 9, 11 };
        int target = 7;
        int index = BinarySearch(arr, target);
        Console.WriteLine(index >= 0 ? $"Found at index {index}" : "Not found");
    }

    static int BinarySearch(int[] arr, int target)
    {
        int left = 0, right = arr.Length - 1;
        while (left <= right)
        {
            int mid = left + (right - left) / 2;
            if (arr[mid] == target) return mid;
            if (arr[mid] < target) left = mid + 1;
            else right = mid - 1;
        }
        return -1;
    }
}
